server/delay_server/src/irodsDelayServer.cpp

```diff
@@ -674,6 +675,47 @@ namespace
 
         return {qstr, job};
     } // make_delay_queue_query_processor
+
+    auto is_current_server_leader() -> bool
+    {
+        std::string hostname;
+        //NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+        char hn[HOST_NAME_MAX + 1]{};
+
+        if (const auto err = gethostname(hn, sizeof(hn)); err != 0) {
+            logger::delay_server::warn("{}: Failed to retrieve local server's hostname. Error {}", __func__, err);
```

Comment:
Getting an error from `gethostname` seems bad, perhaps the log message should be considered an error rather than a warning.

Thoughts?

Reply:
Elsewhere in our code, I see it both as an LOG_ERROR, and a LOG_NOTICE.

I'm leaning ERROR...